nemo/collections/asr/models/ctc_models.py

```diff
@@ -474,11 +474,19 @@ def forward(
         if self.spec_augmentation is not None and self.training:
             processed_signal = self.spec_augmentation(input_spec=processed_signal, length=processed_signal_length)
 
-        encoded, encoded_len = self.encoder(audio_signal=processed_signal, length=processed_signal_length)
+        encoder_output = self.encoder(audio_signal=processed_signal, length=processed_signal_length,)
+        encoded = encoder_output[0]
+        encoded_len = encoder_output[1]
         log_probs = self.decoder(encoder_output=encoded)
         greedy_predictions = log_probs.argmax(dim=-1, keepdim=False)
-
-        return log_probs, encoded_len, greedy_predictions
+        if len(encoder_output) == 2:
+            return (
+                log_probs,
+                encoded_len,
+                greedy_predictions,
+            )
+        else:
```

Comment:
What are these [2] and [3] supposed to be? Add comments, and also update the typecheck signature for output_type